### PR TITLE
fix for #17 add special format

### LINKED ISSUE #17: 3/15 → D/YY
`moment.parseFormat('3/15')` parsed to `D/YY`, I expected: `M/D`

Although D/YY is a valid interpretation, it's not the most intuitive one.

### PR DESCRIPTION
**Pull request by zoepage**
closes #17

### CODE REVIEW

File: tests/moment.parseFormat-test.js [hunk 1]
```diff
@@ -57,3 +57,22 @@ test('GitHub issues', function (t) {
...
+  t.equal(moment.parseFormat('15/15'), 'undefined', 'both numbers are > 12 - 15/15  → undefined')
+  t.equal(moment.parseFormat('03/12'), 'DD/MM', 'both numbers are < 13 - 03/12  → DD/MM')
+  t.equal(moment.parseFormat('3/3'), 'D/M', 'both numbers are < 13 - 3/3  → D/M')
+  t.equal(moment.parseFormat('13/2'), 'DD/M', 'first number is > 12 && <= 31 - 13/2  → DD/M')
+  t.equal(moment.parseFormat('03/12'), 'DD/MM', 'both numbers are < 13 - 03/12 → DD/MM')
+  t.equal(moment.parseFormat('03/03'), 'DD/M', 'both numbers are < 13 - 03/03  → DD/M')
+  t.equal(moment.parseFormat('13/02'), 'DD/M', 'first number is > 12 && <= 31 - 13/02  → DD/M')
+  t.equal(moment.parseFormat('3/12'), 'D/MM', 'both numbers are < 13 - 3/12  → D/MM')
+
+  t.end()
```
Comment (gr2m) [inline tests/moment.parseFormat-test.js]: just a minor: can you remove double spaces before `→` in some of the tests above?

File: tests/moment.parseFormat-test.js [hunk 2]
```diff
@@ -57,3 +57,22 @@ test('GitHub issues', function (t) {
...
 })
+
+test('GitHub issues - special cases', function (t) {
+  // https://github.com/gr2m/moment-parseformat/issues/17
+  t.equal(moment.parseFormat('3/15'), 'M/YY', 'last number is > 12 - 3/15  → M/Y')
+  t.equal(moment.parseFormat('03/15'), 'MM/YY', 'last number is > 12 - 03/15  → MM/Y')
+  t.equal(moment.parseFormat('11/15'), 'MM/YY', 'last number is > 12 - 11/15  → MM/Y')
+
+  t.equal(moment.parseFormat('15/3'), 'DD/M', 'first number is > 12 - 15/3  → DD/M')
+  t.equal(moment.parseFormat('15/15'), 'undefined', 'both numbers are > 12 - 15/15  → undefined')
```
Comment (gr2m) [inline tests/moment.parseFormat-test.js]: should not be `'undefined'` as string, but `undefined` as value

File: tests/moment.parseFormat-test.js [hunk 3]
```diff
@@ -57,3 +57,22 @@ test('GitHub issues', function (t) {
...
+  t.equal(moment.parseFormat('03/12'), 'DD/MM', 'both numbers are < 13 - 03/12  → DD/MM')
+  t.equal(moment.parseFormat('3/3'), 'D/M', 'both numbers are < 13 - 3/3  → D/M')
+  t.equal(moment.parseFormat('13/2'), 'DD/M', 'first number is > 12 && <= 31 - 13/2  → DD/M')
+  t.equal(moment.parseFormat('03/12'), 'DD/MM', 'both numbers are < 13 - 03/12 → DD/MM')
+  t.equal(moment.parseFormat('03/03'), 'DD/M', 'both numbers are < 13 - 03/03  → DD/M')
```
Comment (gr2m) [inline tests/moment.parseFormat-test.js]: test fails, expected is `DD/MM` because of the leading `0`

File: tests/moment.parseFormat-test.js [hunk 4]
```diff
@@ -57,3 +57,22 @@ test('GitHub issues', function (t) {
...
+  t.equal(moment.parseFormat('3/15'), 'M/YY', 'last number is > 12 - 3/15  → M/Y')
+  t.equal(moment.parseFormat('03/15'), 'MM/YY', 'last number is > 12 - 03/15  → MM/Y')
+  t.equal(moment.parseFormat('11/15'), 'MM/YY', 'last number is > 12 - 11/15  → MM/Y')
+
+  t.equal(moment.parseFormat('15/3'), 'DD/M', 'first number is > 12 - 15/3  → DD/M')
+  t.equal(moment.parseFormat('15/15'), 'undefined', 'both numbers are > 12 - 15/15  → undefined')
+  t.equal(moment.parseFormat('03/12'), 'DD/MM', 'both numbers are < 13 - 03/12  → DD/MM')
+  t.equal(moment.parseFormat('3/3'), 'D/M', 'both numbers are < 13 - 3/3  → D/M')
+  t.equal(moment.parseFormat('13/2'), 'DD/M', 'first number is > 12 && <= 31 - 13/2  → DD/M')
+  t.equal(moment.parseFormat('03/12'), 'DD/MM', 'both numbers are < 13 - 03/12 → DD/MM')
```
Comment (gr2m) [inline tests/moment.parseFormat-test.js]: same test as line 69

File: tests/moment.parseFormat-test.js [hunk 5]
```diff
@@ -57,3 +57,22 @@ test('GitHub issues', function (t) {
...
+  t.equal(moment.parseFormat('03/03'), 'DD/M', 'both numbers are < 13 - 03/03  → DD/M')
+  t.equal(moment.parseFormat('13/02'), 'DD/M', 'first number is > 12 && <= 31 - 13/02  → DD/M')
```
Comment (gr2m) [inline tests/moment.parseFormat-test.js]: expected ist `DD/MM`, wegen der `0`

File: moment.parseFormat.js [hunk 6]
```diff
@@ -136,6 +146,27 @@
     // Lets check for 4 digits first, these are years for sure
     format = format.replace(regexYearLong, 'YYYY')
 
+    // check if both numbers are < 13, then it must be DD/MM
+    format = format.replace(regexDayShortMonthShort, 'D/M')
```
Comment (gr2m) [inline moment.parseFormat.js]: comment says DD/MM

File: moment.parseFormat.js [hunk 7]
```diff
@@ -136,6 +146,27 @@
...
+    format = format.replace(regexDayMonth, 'DD/MM')
+
+    // check if first < 10 && last > 12, then it must be M/YY
+    format = format.replace(regexMonthShortYearShort, 'M/YY')
+
+    // check if first < 13 && last > 12, then it must be MM/YY
+    format = format.replace(regexMonthYearShort, 'MM/YY')
+
+    // check if both numbers are < 12, then it must be something wrong
+    format = format.replace(regexDayYear, 'undefined')
```
Comment (gr2m) [inline moment.parseFormat.js]: in that case, I'd just leave the numbers as is

Comment (zoepage) [inline moment.parseFormat.js]: don't understand this :)

Comment (gr2m) [inline moment.parseFormat.js]: if both numbers are > 12, then it's not a valid date, in which case I wouldn't replace it at all

**Comment by gr2m** [comment]
Sorry, I wasn't clear about it. Please remove these parts:

```diff
+    // check if both numbers are < 12, then it must be something wrong
+    format = format.replace(regexDayYear, ‘’)
...
+    if (format.length < 1) {
+      format = undefined
+    }
...
+  t.equal(moment.parseFormat(’15/15’), undefined, ‘both numbers are > 12 - 15/15 → undefined’)
```

If we something can't be turned into a format because it doesn't make sense, like `13/13`, leave the `13/13` in, don't replace anything. Makes sense?

**Comment by zoepage** [comment]
Ok. Will do :)

**Comment by gr2m** [comment]
looks good :+1:  can you squash commits into one docs / test / fix commit please? Please give the test / fix commits a descriptive message, as it will appear on the GitHub releases page, something like `test: parse Day/Month, Month/year #17`, then you can go ahead and merge it

**Pull request merged**